model_compression_toolkit/core/keras/reader/node_builder.py

```diff
@@ -74,23 +74,76 @@ def get_kwargs2index(tfoplambda_layer: TFOpLambda) -> Dict[str, int]:
     Returns:
         A dictionary with argument number and index: {arg_name: arg_index}.
     """
-    kwargs2index = {tf.add: {'x': 0, 'y': 1},
-                    tf.subtract: {'x': 0, 'y': 1},
-                    tf.divide: {'x': 0, 'y': 1},
-                    tf.truediv: {'x': 0, 'y': 1},
-                    tf.multiply: {'x': 0, 'y': 1},
-                    tf.pow: {'x': 0, 'y': 1},
-                    tf.matmul: {'a': 0, 'b': 1}}.get(tfoplambda_layer.function)
-    if not kwargs2index:
-        # In TF 2.15 the function attribute is different and doesn't match the original
-        # operation object we use. Therefore, we extract kwargs2index with the symbol.
-        kwargs2index = {'__operators__.add': {'x': 0, 'y': 1},
-                        'math.add': {'x': 0, 'y': 1},
-                        'math.multiply': {'x': 0, 'y': 1},
-                        'linalg.matmul': {'a': 0, 'b': 1},
-                        'concat': {'values': 0}}.get(tfoplambda_layer.symbol, {})
-
-    return kwargs2index
+
+    full_args = tf_inspect.getfullargspec(tfoplambda_layer.function).args
+
+    return {arg_name: i for i, arg_name in enumerate(full_args)}
+
+
+def _extract_const_attrs_from_kwargs(op_call_kwargs: Dict[str, Any],
+                                     kwarg2index: Dict[str, int],
+                                     weights: Dict[Union[str, int], Any]):
+    """
+    Extract const weights of the layer from the operator's key arguments dictionary.
+    This function extracts the attributes, updates the nodes weights dictionary and removes them from the original
+    kwargs mapping.
+
+    Args:
+        op_call_kwargs: A mapping of the operator key arguments.
+        kwarg2index: A dictionary with argument number and index: {arg_name: arg_index}.
+        weights: Node weights mapping. This dictionary is modified by this function.
+
+    Returns: A modified operator arguments list.
+
+    """
+
+    # read weights from call kwargs
+    weight_keys = []
+    for k, v in op_call_kwargs.items():
+        if is_const(v):
+            if k in kwarg2index:
+                weights.update({kwarg2index[k]: to_numpy(v, is_single_tensor=True)})
+                weight_keys.append(k)
+
+    # remove weights and KerasTensors from op_call_kwargs
```

Comment:
You didn't mention this in the function's description

Reply:
I did: " and removes them from the original kwargs mapping"